src/hooks/search/index.ts

```diff
@@ -218,39 +248,47 @@ const useSearch = (
         }
     }, [router?.isReady]); // Missing query dependency, but that's ok because we only need this on first page load.
 
-    const hasFiltersSet = !!allFilters.length;
+    const hasFiltersSet = !!filters.length;
     const filteredData = (() => {
         if (!data) {
             return [];
         } else if (!hasFiltersSet) {
             return data;
         } else {
             return data.filter(item => {
-                return itemInFilters(item, allFilters);
+                return itemInFilters(item, filters);
             });
         }
     })();
 
-    const numberOfResults = filteredData.length;
-    const shownData = filteredData.slice(0, resultsToShow);
-    const fullyLoaded = resultsToShow >= numberOfResults;
+    const clearAll = () => {
+        setSearchString('');
+        setFilters([]);
+        setSortBy('Most Recent');
+        updatePageMeta();
+    };
 
     return {
-        data: shownData,
-        error,
-        isValidating,
-        resultsToShow,
-        setResultsToShow,
-        allFilters,
-        setAllFilters,
-        onSearch: debouncedOnSearch,
-        onFilter,
-        searchString,
-        setSearchString,
-        fullyLoaded,
-        numberOfResults,
-        onSort,
-        sortBy,
+        searchBoxProps: {
```

Comment:
Put a lot of thought into the best way to refactor the search so it could be used on the topic pages as well as content type/sitewide search. The crux of the problem is to find a way where the search logic could be centralized but applied to multiple components that might be scattered throughout the page. This seems to be the best way, which was similar to the old implementation, but I thought packaging the props by component is probably the cleanest so that they can later be spread into each one easily. Also, the case in which a component handles multiple search functionalities (e.g. mobile filter component handling sorting & filtering) becomes much more declarative -- you just need to spread in the related prop group (`{...sortBoxProps} {...filterProps}`) 

Also, by not relying on one centralized search component, and rather using `useSearch` in combination with whichever specific search related components are needed, it removes a lot of the need for excessive prop passing and one-off if-statements.